### PR TITLE
[INLONG-10225][Audit] Audit SDK provides the ability to automatically assign and manage Audit ID

### LINKED ISSUE #10225: [Improve][Audit] Audit SDK provides the ability to automatically assign and manage Audit ID
### Description

Audit SDK provides the ability to automatically assign and manage Audit ID

### InLong Component

InLong Audit

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by doleyzi**
- Fixes #10225

### Motivation

There are more and more Audit modules, requiring Audit SDK for unified distribution and management.

### Modifications

Audit SDK provides the ability to automatically assign and manage Audit IDs.

### CODE REVIEW

**Review by justinwwhuang — DISMISSED** [review]
LGTM

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+ */
+
+package org.apache.inlong.audit;
+
+/**
+ * Audit item management, each module is assigned two baseline audit item IDs, namely receiving and sending.
+ */
+public enum AuditIdEnum {
+
+    SDK_INPUT(1, "Audit for SDK to receive data"),
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java:25]:
```suggestion
    SDK_INPUT(1, "Number of Data Records Received for SDK"),
```

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    SDK_OUTPUT(2, "Audit for SDK to send data"),
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java:26]:
```suggestion
    SDK_OUTPUT(2, "Number of Data Records Sent for SDK"),
```

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java [hunk 3]
```diff
@@ -56,8 +57,10 @@ public class AuditReporterImpl implements Serializable {
     private static final String FIELD_SEPARATORS = ":";
     private static final long DEFAULT_AUDIT_VERSION = -1;
     private static final int BATCH_NUM = 100;
-    private final ReentrantLock GLOBAL_LOCK = new ReentrantLock();
     private static final int PERIOD = 1000 * 60;
+    // Resource isolation key is used in checkpoint and other scenarios.DEFAULT 0.
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java:61]: Resource isolation key is used in checkpoint, the default value is 0

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java [hunk 4]
```diff
@@ -502,4 +501,22 @@ private void checkFlushTime() {
             }
         });
     }
+
+    /**
+     * Generate success and failure, real-time and non-real-time, retry, discard and other Audit item IDs through the baseline Audit ID.
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java:506]:
```suggestion
     * Generate Audit item IDs.
```

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/util/AuditManagerUtils.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    private static String buildAuditIdSuffix(int auditId) {
+        StringBuilder auditIdBinaryString = new StringBuilder(Integer.toBinaryString(auditId));
+        for (int i = auditIdBinaryString.length(); i < AUDIT_SUFFIX_LENGTH; i++) {
+            auditIdBinaryString.insert(0, "0");
+        }
+        return auditIdBinaryString.toString();
+    }
+
+    /**
+     * Generate success and failure, real-time and non-real-time, retry, discard and other Audit item IDs through the baseline Audit ID.
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/util/AuditManagerUtils.java:58]:
```suggestion
     * Generate Audit item IDs.
```

File: inlong-audit/audit-sdk/README.md [hunk 6]
```diff
@@ -11,7 +11,7 @@ The Audit Sdk uses log production time as the audit standard,
 which can ensure that each module is reconciled in accordance with the unified audit standard.
 
 ## Usage
-### setAuditProxy API
+### SetAuditProxy API
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/README.md:14]:
```suggestion
### Configure Audit Proxy Addresses
```

File: inlong-audit/audit-sdk/README.md [hunk 7]
```diff
@@ -11,7 +11,7 @@ The Audit Sdk uses log production time as the audit standard,
...
 Set the AuditProxy ip:port list. The Audit Sdk will summarize the results according to the cycle
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/README.md:15]:
```suggestion
The Audit SDK will summarize the results according to the cycle 
```

File: inlong-audit/audit-sdk/README.md [hunk 8]
```diff
@@ -68,4 +76,20 @@ In order to ensure the accuracy of auditing, each operator needs to create an au
...
+
+#### Build Audit item ID
+```java
+        AuditManagerUtils.buildAuditId(AuditIdEnum baseAuditId,
+        boolean success,
+        boolean isRealtime,
+        boolean discard,
+        boolean retry);
+```
+| parameter       | meaning                                                                      |
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/README.md:89]:
```suggestion
| parameter       | description                                                                     |
```

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java (hunk 1 above)
Comment (dockerzhang) [reply inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java:25]: Received Audit Metrics for SDK

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java (hunk 2 above)
Comment (dockerzhang) [reply inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java:26]: ReSent Audit Metrics for SDK

Comment (dockerzhang) [reply inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java:26]: Sent Audit Metrics for SDK

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java [hunk 9]
```diff
@@ -502,4 +501,22 @@ private void checkFlushTime() {
...
+     * @param retry
+     * @return
+     */
+    public int buildAuditId(AuditIdEnum baseAuditId,
+            boolean success,
+            boolean isRealtime,
+            boolean discard,
+            boolean retry) {
+        return AuditManagerUtils.buildAuditId(baseAuditId, success, isRealtime, discard, retry);
+    }
```
Comment (aloyszhang) [inline inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java:521]: Why add a method that just calls a public static method `AuditManagerUtils.buildAuditId`?

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java (hunk 1 above)
Comment (doleyzi) [reply inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java]: done

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java (hunk 2 above)
Comment (doleyzi) [reply inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditIdEnum.java]: done

File: inlong-audit/audit-sdk/README.md (hunk 6 above)
Comment (doleyzi) [reply inlong-audit/audit-sdk/README.md]: done

File: inlong-audit/audit-sdk/README.md (hunk 7 above)
Comment (doleyzi) [reply inlong-audit/audit-sdk/README.md]: done

File: inlong-audit/audit-sdk/README.md (hunk 8 above)
Comment (doleyzi) [reply inlong-audit/audit-sdk/README.md]: done

**Review by justinwwhuang — APPROVED** [review]

File: inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java (hunk 9 above)
Comment (aloyszhang) [reply inlong-audit/audit-sdk/src/main/java/org/apache/inlong/audit/AuditReporterImpl.java:521]: Make sense

File: inlong-audit/audit-sdk/README.md [hunk 10]
```diff
@@ -22,38 +22,46 @@ If the AuditProxy changes in real time, then the business program needs to call
     AuditOperator.getInstance().setAuditProxy(ipPortList);
 ```
 
-### add API
+### Description API for Add Audit Data
```
Comment (dockerzhang) [inline inlong-audit/audit-sdk/README.md:25]:
```suggestion
### Add Audit Data
```

**Review by dockerzhang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**